router/ubios/setup.go

```diff
@@ -1,159 +1,75 @@
 package ubios
 
 import (
-	"bufio"
-	"errors"
+	"bytes"
 	"fmt"
 	"net"
 	"os"
 	"os/exec"
-	"strings"
 
 	"github.com/nextdns/nextdns/config"
+	"github.com/nextdns/nextdns/router/internal"
 )
 
 type Router struct {
-	LANIPv6   string
-	UsePodman bool
+	DNSMasqPath     string
+	ListenPort      string
+	ClientReporting bool
 }
 
 func New() (*Router, bool) {
 	if st, _ := os.Stat("/data/unifi"); st == nil || !st.IsDir() {
 		return nil, false
 	}
-	ipv6, _ := getIface6GlobalIP("br0")
-	usePodman, _ := isContainerized()
 	return &Router{
-		LANIPv6:   ipv6,
-		UsePodman: usePodman,
+		DNSMasqPath: "/run/dnsmasq.conf.d/znextdns.conf",
+		ListenPort:  "5342",
 	}, true
 }
 
-func isContainerized() (bool, error) {
-	f, err := os.Open("/proc/1/cgroup")
-	if err != nil {
-		return false, err
-	}
-	defer f.Close()
-	s := bufio.NewScanner(f)
-	for s.Scan() {
-		flds := strings.Split(s.Text(), ":")
-		if len(flds) != 3 {
-			continue
-		}
-		if flds[2] != "/" && flds[2] != "/init.scope" {
-			return true, nil
-		}
-	}
-	return false, nil
+func (r *Router) Configure(c *config.Config) error {
+	c.Listens = []string{net.JoinHostPort("localhost", r.ListenPort)}
+	r.ClientReporting = c.ReportClientInfo
+	// Domain-based traffic management rules depend on dnsmasq handling requests. So forcefully disable caching.
+	c.CacheSize = "0"
```

Comment:
Keeping dnsmasq cache in front of the CLI would remove the ability to use conditional profiles, as you need to maintain a segmented cache per profile.

Reply:
Ah yes indeed. I bet caching in dnsmasq can be disabled in the config, and then nextdns can handle caching.